pkg/operator/sync.go

```diff
@@ -1261,6 +1261,16 @@ func (optr *Operator) reconcileMachineOSBuilder(mob *appsv1.Deployment) error {
 		return fmt.Errorf("could not reconcile global pull secret copy: %w", err)
 	}
 
+	// Set the owner ref of the machine-os-builder deployment to the MOSC
+	if len(layeredMCPs) > 0 {
+		mosc, err := build.GetMachineOSConfigForPool(optr.client, layeredMCPs)
+		if err != nil {
+			return err
+		}
+		oref := metav1.NewControllerRef(mosc, mcfgv1.SchemeGroupVersion.WithKind("MachineOSConfig"))
```

Comment:
question (non-blocking): Would we want the MachineOSConfig to be deleted if the `machine-os-builder` deployment is deleted? I'm not sure that it makes sense for us to delete a user-supplied object in this particular case.

Reply:
The `machine-os-builder` deployment has the MOSC as the owner, so if the deployment is delete the MOSC will not be deleted. But if the MOSC is deleted, it will ensure that the deployment is always deleted as that is what we want right?

Reply:
It makes sense for total MOSC deletion to cause the deployment to be deleted. What I don't know about is whether that can express an all-or-nothing relationship. In other words, if there are multiple MOSCs and one is deleted, would that delete the deployment? Or would that only happen when all of them are deleted?

Reply:
So I was seeing some cases online of objects having multiple owners and being deleted only if all the owners have been deleted. However, while trying this with the deployment, it doesn't seem to add multiple owners to it. I have removed all ownerReference for the deployment as it is cleared out by the build controller anyways.